beater/beater.go

```diff
@@ -6,15 +6,18 @@ import (
 	"net"
 	"net/http"
 	"regexp"
+	"sync"
 
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
 type beater struct {
-	config *Config
-	server *http.Server
+	config  *Config
+	mutex   sync.RWMutex // guards server and stopped
```

Comment:
no need for a RWMutex if `RLock` and `RUnlock` isn't used.

Reply:
true, good point